docs/getting-started/contributing/contributing-with-git.md

```diff
@@ -0,0 +1,196 @@
+# Contributing with Git
+
+Community contributions are an important part of USGS open source projects.  Contributions from people like you help make our projects at the USGS better for everyone.  And, by contributing, you can build your skills, display your work, and form community connections.
+
+This guide walks through the process of using git.  Git is the system the USGS uses to manage changes and contributions to our software projects.
+
+If you aren't comfortable contributing code, you can also help by contributing documentation or tutorials, or [opening](../../how-to-guides/software-management/guidelines-for-reporting-issues.md) and commenting on git issues.
+
```

Comment:
The flow chart below is pretty technical. Then the prerequisites are high level, conceptual, mixed with technical. I wonder if some additional text here, before the flow chart might make sense. For example: 

'In order to contribute to one of our repositories, you will be using `git` and GitHub. `git` is a tool, installed on your computer that helps you version control software. GitHub is a remote code hosting company and website where developers and users collaborate. 

There are a few concepts that are important to understand before getting into the technical details. These are:

- What are issues?
- What is a fork?
- What is a clone?
- What are branches?
- What is a remote?

GitHub provides terrific documentation on each of these topics which we link to below.

Reply:
And then get into the technical. 

- Now that you know what an issue is, let's open one on a repo.
- Now that you know what a fork is, here are there docs for how to go about forking. Let's fork a repo.
- etc...

If it isn't clear I am struggling with the overall organization because this document is conceptual and practical. Is it possible to split these (inside this document at first?). Does that even make sense?

Reply:
My thought process in writing the doc as a whole, is that someone can go down the list, and they will get the technical things they need to do, as well as a (minimal) conceptual explanation.

I can sort of understand that you want to separate the conceptual and practical, but my opinion is that it's helpful to see them together.

Reply:
The git/GitHub dichotomy is important - I'll see if I can explain that better.  Are our public open source projects all on GitHub?  Are some on GitLab too, or is that only internal ones?

Reply:
Some public stuff is on GitLab. In fact, the repository of record is officially GitLab. Personally, I would not discuss GitLab at all to avoid confusion.

On the conceptual vs. technical. Could you label / use collapsing / some other formatting trick to help separate? I 100% get where you are coming from wanting to keep them together. My concern is related to the firehose of information that this ends up being, but that might be necessary in this case. Is the target audience someone who has never used git or someone who uses git but wants to learn the specifics of contributing to our repos (for example, stuff like CI is a little different than a vanilla, uses GitHub CI).

Reply:
The target I've been writing for is someone who hasn't used git (at least not much).  CI is beyond the scope of this doc.

Reply:
I do think it's a good idea to link out to GitHub for more details on the conceptual stuff.  I still want a minimum explanation here though.  I'll see what I can do, maybe some formatting will help.

Reply:
> The target I've been writing for is someone who hasn't used git (at least not much).

If this is for someone completely new to git, why not start them by linking to GitHub's [Hello World](https://docs.github.com/en/get-started/start-your-journey/hello-world) and have them work through that as a prerequisite?

Reply:
I moved some of the conceptual parts to collapsed boxes, the main flow of the doc should be a bit more technical/instructional now.

Reply:
The GitHub Hello World does a lot of the same things as this document (minus forking). I was under the impression we wanted something similar here in the Astro docs.